src/inputType/PinchInput.ts

```diff
@@ -114,16 +115,23 @@ export class PinchInput implements IInputType {
 	}
 
   private onPinchStart(event) {
-    this._prev = event.scale;
-    this.observer.hold(this, event);
+		this._prev = 1;
+		this._base = this.observer.get(this)[this.axes[0]];
+		const offset = this._base * (event.scale - this._prev) * this.options.scale;		
+		this.observer.hold(this, event);
+		this.observer.change(this, event, toAxis(this.axes, [offset]));
+		this._prev = event.scale;		
   }
   private onPinchMove(event) {
-    const offset = (event.scale - this._prev) * this.options.scale;
-    this.observer.change(this, event, toAxis(this.axes, [offset]));
+		const offset = this._base * (event.scale - this._prev) * this.options.scale;		
+		this.observer.change(this, event, toAxis(this.axes, [offset]));
     this._prev = event.scale;
   }
   private onPinchEnd(event) {
+		const offset = this._base * (event.scale - this._prev) * this.options.scale;		
```

Comment:
How about create private function?
offset code is duplicated at onPinchStart/Move/End.

```ts
private getOffset(prev: number = 1) {
    return this._base * (event.scale - prev) * this.options.scale;	
}
```